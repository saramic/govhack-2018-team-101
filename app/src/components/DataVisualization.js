import React from "react";
import { Modal, Button } from "antd";

const DataVisualization = ({ data }) => {
  console.log(data);
  return (
    <div className="visualization">
      {data.visualisation ? (
        <img src={"/images/viz/" + data.visualisation} />
      ) : null}

      <p>{data.name}</p>
      <p>
        <a href={data.hackerspace_id} target="_blank">
          Hackerspace dataset
        </a>
      </p>
      <p>
        <a href={data.link} target="_blank">
          Link to data
        </a>
      </p>

      {/* <Modal
        title={null}
        visible={}
        footer={null}
      >IMAGE HERE</Modal> */}
    </div>
  );
};

export default DataVisualization;
