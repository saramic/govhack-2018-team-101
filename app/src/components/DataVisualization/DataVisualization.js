import React from "react";

const DataVisualization = ({ data }) => {
  return (
    <div className="visualization">
      {data.visualisation ? (
        <img src={"/images/viz/" + data.visualisation} />
      ) : null}
      <h1>{data.name}</h1>
      <p>
        <a href={data.link} target="_blank">
          Original Data Source
        </a>
      </p>
      <p>
        <a href={data.hackerspace_id} target="_blank">
          GovHack 2018 Hackerspace Info
        </a>
      </p>
    </div>
  );
};

export default DataVisualization;
