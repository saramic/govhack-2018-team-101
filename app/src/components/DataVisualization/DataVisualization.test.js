import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import DataVisualization from './DataVisualization';
// following 3 lines should go in a setupEnzyme.js but need to `yarn run eject`
// to allow the overrides required in package.json
// "jest": {
//   "setupTestFrameworkScriptFile": "<rootDir>src/setupEnzyme.js"
// }
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
// END section for setup Enzyme.js

describe('DataVisualization', () => {
  it('should render correctly WITHOUT visualization image', () => {
    const output = shallow(
      <DataVisualization data={
        {
          name: 'The Data Name',
          link: 'http://data.source/link',
          hackerspace_id: '999',
        }
      }/>
    )
    expect(shallowToJson(output)).toMatchSnapshot();
  })
  it('should render correctly WITH visualization image', () => {
    const output = shallow(
      <DataVisualization data={
        {
          visualisation: 'the_image_url',
          name: 'The Data Name',
          link: 'http://data.source/link',
          hackerspace_id: '999',
        }
      }/>
    )
    expect(shallowToJson(output)).toMatchSnapshot();
  })
})
