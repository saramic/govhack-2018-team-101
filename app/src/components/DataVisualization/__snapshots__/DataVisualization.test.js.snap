// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataVisualization should render correctly WITH visualization image 1`] = `
<div
  className="visualization"
>
  <img
    src="/images/viz/the_image_url"
  />
  <h1>
    The Data Name
  </h1>
  <p>
    <a
      href="http://data.source/link"
      target="_blank"
    >
      Original Data Source
    </a>
  </p>
  <p>
    <a
      href="999"
      target="_blank"
    >
      GovHack 2018 Hackerspace Info
    </a>
  </p>
</div>
`;

exports[`DataVisualization should render correctly WITHOUT visualization image 1`] = `
<div
  className="visualization"
>
  <h1>
    The Data Name
  </h1>
  <p>
    <a
      href="http://data.source/link"
      target="_blank"
    >
      Original Data Source
    </a>
  </p>
  <p>
    <a
      href="999"
      target="_blank"
    >
      GovHack 2018 Hackerspace Info
    </a>
  </p>
</div>
`;
